### DISCUSSION TITLE
Getting the test step execution time as part of the ExecutionContext

### DISCUSSION THREAD

**Discussion by declancarrollcb** [getgauge/gauge#1995] [General]
Hi,

Does anyone know of a way to get the test run time in a hook? I was hoping to export some of the test step run times to determine areas that might need improving.

**Comment by sriv** [comment]
hi @declancarrollcb - the execution time is not passed to the hooks at the moment. It is available to the plugins though, so if you were to use the json-report, you should be able to get execution times of the steps.

**Reply by declancarrollcb** [reply]
Thanks @sriv. Would it be worth creating an issue/feature request to get the time added to the hooks? It would be something I would be interested in. If the work required would be small enough I'd be happy to tackle it myself.

**Reply by sriv** [reply]
I think this information is worth including in the execution context. Please create an issue and we can discuss ideas around this.

**Reply by shortstacked** [reply]
I've added issue #2010 to capture it. Thanks.
